### ISSUE TITLE
Nice interface for doing the training steps

### ISSUE THREAD

**Issue by mscroggs** [UCL/causalprog#137]
- [ ] Given a training set, learn $\hat{r}$ from (z, x, y, l) data. "For instance, XGBoost, random forest and TabPFN etc can be used"
- [ ] Fit a normalising flow to get $\hat\theta_x$ from training data
